src/amber/controller/helpers/responders.cr

```diff
@@ -61,9 +66,12 @@ module Amber::Controller::Helpers
     end
 
     def set_response(body, status_code = 200, content_type = Content::TYPE[:html])
-      context.response.status_code = status_code
+      if body.is_a?(Proc)
+        body = body.call
+      end
+      context.response.status_code = status_code if context.response.status_code == 200
       context.response.content_type = content_type
-      context.content = body
+      context.content = body.is_a?(String) ? body : body.to_s
```

Comment:
By the time it gets into the `set_response` method body should already be a string.

Reply:
see above

Reply:
Same as my response above.